internal/operators/ocs/ocs_operator.go

```diff
@@ -146,42 +160,49 @@ func (o *operator) GetHostRequirements(_ context.Context, cluster *common.Cluste
 	}
 
 	role := common.GetEffectiveRole(host)
-	if numOfHosts <= 3 { // Compact Mode
+	label := host.Labels
+
+	if numOfHosts <= o.config.OCSNumMinimumHosts { // Compact mode
 		var reqDisks int64 = 1
 		if diskCount > 0 {
 			reqDisks = diskCount
 		}
-		// for each disk ocs requires 2 CPUs and 5 GiB RAM
-		if role == models.HostRoleMaster || role == models.HostRoleAutoAssign {
+		if label == ocsLabel && (role == models.HostRoleMaster || role == models.HostRoleAutoAssign) {
 			return &models.ClusterHostRequirementsDetails{
 				CPUCores: o.config.OCSPerHostCPUCompactMode + (reqDisks * o.config.OCSPerDiskCPUCount),
 				RAMMib:   conversions.GibToMib(o.config.OCSPerHostMemoryGiBCompactMode + (reqDisks * o.config.OCSPerDiskRAMGiB)),
 			}, nil
+		} else if label == ocsLabel && role == models.HostRoleWorker {
+			return &models.ClusterHostRequirementsDetails{
+				CPUCores: o.config.OCSPerHostCPUStandardMode + (reqDisks * o.config.OCSPerDiskCPUCount),
+				RAMMib:   conversions.GibToMib(o.config.OCSPerHostMemoryGiBStandardMode + (reqDisks * o.config.OCSPerDiskRAMGiB)),
+			}, nil
+		} else { // label != ocsLabel
+			return &models.ClusterHostRequirementsDetails{CPUCores: 0, RAMMib: 0}, nil
+		}
+	} else { // Standard mode
+		if label != ocsLabel || role == models.HostRoleMaster { // In standard mode, OCS does not run on master nodes so return zero
+			return &models.ClusterHostRequirementsDetails{CPUCores: 0, RAMMib: 0}, nil
+		} else { // label == ocsLabel && (role == models.HostRoleWorker || role == models.HostRoleAutoAssign)
+			return &models.ClusterHostRequirementsDetails{
+				CPUCores: o.config.OCSPerHostCPUStandardMode + (diskCount * o.config.OCSPerDiskCPUCount),
+				RAMMib:   conversions.GibToMib(o.config.OCSPerHostMemoryGiBStandardMode + (diskCount * o.config.OCSPerDiskRAMGiB)),
+			}, nil
 		}
-		// regular worker req
-		return &models.ClusterHostRequirementsDetails{
-			CPUCores: o.config.OCSPerHostCPUStandardMode + (reqDisks * o.config.OCSPerDiskCPUCount),
-			RAMMib:   conversions.GibToMib(o.config.OCSPerHostMemoryGiBStandardMode + (reqDisks * o.config.OCSPerDiskRAMGiB)),
-		}, nil
-	}
-
-	// In standard mode, OCS does not run on master nodes so return zero
-	if role == models.HostRoleMaster {
-		return &models.ClusterHostRequirementsDetails{CPUCores: 0, RAMMib: 0}, nil
 	}
 
-	// worker and auto-assign
-	if diskCount > 0 {
-		// for each disk ocs requires 2 CPUs and 5 GiB RAM
-		return &models.ClusterHostRequirementsDetails{
-			CPUCores: o.config.OCSPerHostCPUStandardMode + (diskCount * o.config.OCSPerDiskCPUCount),
-			RAMMib:   conversions.GibToMib(o.config.OCSPerHostMemoryGiBStandardMode + (diskCount * o.config.OCSPerDiskRAMGiB)),
-		}, nil
-	}
-	return &models.ClusterHostRequirementsDetails{
-		CPUCores: o.config.OCSPerHostCPUStandardMode,
-		RAMMib:   conversions.GibToMib(o.config.OCSPerHostMemoryGiBStandardMode),
-	}, nil
+	//// worker and auto-assign
```

Comment:
Why do we want to comment this part?

Reply:
The commented-out is no longer of use however I have just kept it there because I want to confirm one thing with Rewant first before completely removing it